src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsPathContextProvider.cs

```diff
@@ -310,5 +312,63 @@ public IVsPathContext GetSolutionPathContext()
                 pathContext.FallbackPackageFolders.Cast<string>(),
                 trie);
         }
+
+        private IList<Project> GetProjectsInSolution(DTE dte)
+        {
+            var supportedProjects = dte.Solution.Projects;
+            Projects projects = dte.Solution.Projects;
+
+            List<Project> projectList = new List<Project>();
+            var item = projects.GetEnumerator();
+            while (item.MoveNext())
+            {
+                var project = item.Current as Project;
+                if (project == null)
+                {
+                    continue;
+                }
+
+                if (project.Kind == ProjectKinds.vsProjectKindSolutionFolder)
+                {
+                    IEnumerable<Project> solutionFolderProjects = GetSolutionFolderProjects(project);
+                    projectList.AddRange(solutionFolderProjects);
+                }
+                else
+                {
+                    projectList.Add(project);
+                }
+            }
+
+            return projectList;
+        }
+
+        /// <summary>
+        /// Gets all projects that are not solution folders within the given solutionFolder.
+        /// (i.e. If the hierarchy is src\a\b.xproj, src\c.xproj; the resulting list will contain b.xproj and c.xproj).
+        /// </summary>
+        private IEnumerable<Project> GetSolutionFolderProjects(Project solutionFolder)
+        {
+            List<Project> projectList = new List<Project>();
+            for (var i = 1; i <= solutionFolder.ProjectItems.Count; i++)
+            {
+                var subProject = solutionFolder.ProjectItems.Item(i).SubProject;
```

Comment:
I think array index `i` should start from `0` otherwise
- When `i == solutionFolder.ProjectItems.Count` then `solutionFolder.ProjectItems.Item(i)` will throw null reference exception.
- `ProjectItems.Item(0)` will never be read.

I am not sure If i missing anything here.

Reply:
I assume that the API uses 1-based indexing, not 0-based indexing, otherwise Donnie would have noticed his VS crash when he tested this.

Reply:
May be `foreach` is a better choice [see doc](https://docs.microsoft.com/en-us/dotnet/api/envdte.projectitems?view=visualstudiosdk-2017)

Reply:
@kartheekp-ms good find, but I believe this ProjectItems](https://docs.microsoft.com/en-us/dotnet/api/envdte.projectitems?view=visualstudiosdk-2017&viewFallbackFrom=visualstudiosdk-2019) collection is not implemented as an array, but instead, a dictionary. Why item 0 is not used is not explained by my quick search.